src/test/java/timetables_sort_use_case/application_business/AllTimetablesInteractorTest.java

```diff
@@ -0,0 +1,66 @@
+package timetables_sort_use_case.application_business;
+
+import entities.Timetable;
+import entities.TimetableCourse;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+
+import java.util.ArrayList;
+import java.util.concurrent.Flow;
+import java.util.List;
+
+/**
+ * A suite of tests to confirm that AllTimetablesPublisher properly updates all its subscribers
+ */
+public class AllTimetablesInteractorTest {
+
+    private AllTimetablesInteractor publisher;
+    private TestSubscriberUser User;
+    private Flow.Subscriber<Object> subscriber;
+    private Timetable emptyTimetable;
+
+    @BeforeEach
+    void setUp(){
+         this.publisher = new AllTimetablesInteractor();
+         this.subscriber = new Flow.Subscriber<>() {
+             @Override
+             public void onSubscribe(Flow.Subscription subscription) {
+
+             }
+
+             @Override
+             public void onNext(Object item) {
+                 User.timetable = (Timetable) item;
+             }
+
+             @Override
+             public void onError(Throwable throwable) {
+
+             }
+
+             @Override
+             public void onComplete() {
+
+             }
+         };
+         this.User = new TestSubscriberUser(subscriber);
+         ArrayList<TimetableCourse> emptyCourselist = new ArrayList<>();
+         Timetable emptyTimetable = new Timetable(emptyCourselist, "F");
+         List<Timetable> updatedTimetables = new ArrayList<>();
+         updatedTimetables.add(emptyTimetable);
+         publisher.onNext(updatedTimetables);
+         this.emptyTimetable = emptyTimetable;
+    }
+
+    /**
```

Comment:
This is really rigorous, even testing subscribers. 😳.

I have generally thought this would be more of an integration test thing, because the individual methods don't do too much, (and are really simple, most onNext methods are basically setters) but I guess one could view this as a little integration test :p. I'm not sure if it;s fully necessary though. Should I integrate it into mine?

Reply:
Im not sure so i played it safe